src/log.ts

```diff
@@ -112,6 +116,31 @@ class Log implements LogSeverityFunctions {
      * @type {string}
      */
     this.name = this.formattedName_.split('/').pop()!;
+    this.jsonFieldsToTruncate = [
+      // Winston:
+      'jsonPayload.fields.metadata.structValue.fields.stack.stringValue',
+      // Bunyan:
+      'jsonPayload.fields.msg.stringValue',
+      'jsonPayload.fields.err.structValue.fields.stack.stringValue',
+      'jsonPayload.fields.err.structValue.fields.message.stringValue',
+      // All:
+      'jsonPayload.fields.message.stringValue',
+    ];
+
+    // Prepend all custom fields to be truncated to a list with defaults, thus
+    // custom fields will be truncated first. Make sure to filter out fields
+    // which are not in EntryJson.jsonPayload
+    if (options.jsonFieldsToTruncate !== undefined) {
```

Comment:
what supposed to happen if `jsonFieldsToTruncate` is passed as `null`?

Reply:
Yep, fixed - thanks!